src/service/ReservationService.java

```diff
@@ -0,0 +1,101 @@
+package service;
+
+import model.Customer;
+import model.IRoom;
+import model.Reservation;
+
+import java.util.*;
+
+public class ReservationService {
+    public static final ReservationService reservationService = new ReservationService();
+    private static final List<Reservation> reservations = new ArrayList<>();
+    private static final Map<String, IRoom> rooms = new HashMap<>();
+
+    public Collection<IRoom> findRooms(Date checkInDate, Date checkOutDate) {
+        List<IRoom> availableRooms = new ArrayList<>();
+        for (IRoom room : rooms.values()) {
+            if (reservationService.checkAvailability(room, checkInDate, checkOutDate)) {
```

Comment:
This is very nice that you delegate the `checkAvailability`  check to a `reservationService`!